### ISSUE TITLE
False positive type error when combining Into and Box

### ISSUE THREAD

**Issue by Exidex** [intellij-rust/intellij-rust#10082]
## Environment

* **IntelliJ Rust plugin version:** 0.4.187.5175-223
* **Rust toolchain version:** 1.67.0 (fc594f156 2023-01-24) x86_64-unknown-linux-gnu
* **IDE name and version:** CLion 2022.3.2 (CL-223.8617.54)
* **Operating system:** Linux 5.15.0-58-generic
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.macros.proc

## Problem description
The following code fragment shows error in Intellij but compiles fine using cargo.
Couldn't reproduce without a `svg` crate, so I don't know the root cause and not sure it there's already an issue for this.

## Steps to reproduce

Cargo.toml:
```toml
[package]
name = "test"
version = "0.1.0"
edition = "2021"

[dependencies]
svg = "0.13.0"
```

main.rs:
```rust
use svg::Node;
use svg::node::element::Rectangle;

fn main() {
    let node = Rectangle::new();
    let element = node.into();
    let _vec: Vec<Box<dyn Node>> = vec![element];
    //                             ^^^^^^^^^^^^^
    // mismatched types [E0308]
    // expected `Vec<Box<dyn Node>>`, found `Vec<Element>`
}
```
